### PR TITLE
OBMIN-315 removed dto

### PR DESCRIPTION
**Pull request by ImAndVlad**
- deleted DTO UserChangeEmailDto
- refacted logic in service
- refacted methods in controller
- refacted tests

### CODE REVIEW

File: src/main/java/space/obminyashka/items_exchange/service/UserService.java [hunk 1]
```diff
@@ -41,20 +41,19 @@ public interface UserService {
...
-     * Update password of an existed user with new one
-     * @param userChangePasswordDto DTO which contains old password and doubled new password
-     * @param user existed user to update
-     * @return a message as the result of the operation
+     * Update the user in the database.
+     *
+     * @param user The user object to be updated.
      */
-    String updateUserPassword(UserChangePasswordDto userChangePasswordDto, User user);
+    void update(User user);
```
Comment (ennnbaev) [inline src/main/java/space/obminyashka/items_exchange/service/UserService.java:48]: It would be better to rename this method

**Review by ennnbaev — DISMISSED** [review]
Good job!

File: src/main/java/space/obminyashka/items_exchange/controller/UserController.java [hunk 2]
```diff
@@ -108,18 +110,15 @@ public String updateUserPassword(@Valid @RequestBody UserChangePasswordDto userC
...
-            throw new DataConflictException(getMessageSource(ResponseMessagesHandler.ExceptionMessage.EMAIL_OLD));
-        }
-        if (userService.existsByEmail(userChangeEmailDto.getNewEmail())) {
-            throw new DataConflictException(getMessageSource(
-                    ResponseMessagesHandler.ValidationMessage.DUPLICATE_EMAIL));
-        }
-
-        return userService.updateUserEmail(userChangeEmailDto, user);
+    public String updateUserEmail(@Parameter(name = "New email")
+                                  @Email(regexp = PatternHandler.EMAIL, message = "{" + ResponseMessagesHandler.ValidationMessage.INVALID_EMAIL + "}")
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/controller/UserController.java:114]: You should statically import INVALID_EMAIL and that would be looked a way better.

File: src/main/java/space/obminyashka/items_exchange/controller/UserController.java [hunk 3]
```diff
@@ -219,4 +218,16 @@ private User getUser(String username) {
         return userService.findByUsernameOrEmail(username).orElseThrow(() -> new UsernameNotFoundException(
                 getMessageSource(ResponseMessagesHandler.ExceptionMessage.USER_NOT_FOUND)));
     }
+
+    private User findUserByValidEmail(Authentication authentication, String email) throws DataConflictException {
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/controller/UserController.java:222]: That method ruins the Single responsibility principle - it does more than 1 general operation and should be split into a few others:
1. It validates email
2. It returns a user

**Review by rpkyrych — CHANGES_REQUESTED** [review]
Split the new controller's method according to Single Responsibility principal

File: src/main/java/space/obminyashka/items_exchange/controller/UserController.java (hunk 3 above)
Comment (ImAndVlad) [reply src/main/java/space/obminyashka/items_exchange/controller/UserController.java]: I splited this method

File: src/main/java/space/obminyashka/items_exchange/controller/UserController.java (hunk 2 above)
Comment (ImAndVlad) [reply src/main/java/space/obminyashka/items_exchange/controller/UserController.java:114]: If I statically import INVALID_EMAIL it breaks the build

Comment (ImAndVlad) [reply src/main/java/space/obminyashka/items_exchange/controller/UserController.java]: refacted

File: src/main/java/space/obminyashka/items_exchange/controller/UserController.java [hunk 4]
```diff
@@ -108,18 +111,16 @@ public String updateUserPassword(@Valid @RequestBody UserChangePasswordDto userC
...
+                                  @Parameter(hidden = true) Authentication authentication) throws DataConflictException {
         User user = getUser(authentication.getName());
-        if (user.getEmail().equals(userChangeEmailDto.getNewEmail())) {
-            throw new DataConflictException(getMessageSource(ResponseMessagesHandler.ExceptionMessage.EMAIL_OLD));
-        }
-        if (userService.existsByEmail(userChangeEmailDto.getNewEmail())) {
-            throw new DataConflictException(getMessageSource(
-                    ResponseMessagesHandler.ValidationMessage.DUPLICATE_EMAIL));
-        }
+        validEmail(user.getEmail(), email);
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/controller/UserController.java:119]: How about **checkEmailUniqueAndNotUsed**?

**Review by rpkyrych — DISMISSED** [review]

**Review by ennnbaev — APPROVED** [review]

File: src/main/java/space/obminyashka/items_exchange/controller/UserController.java (hunk 4 above)
Comment (ImAndVlad) [reply src/main/java/space/obminyashka/items_exchange/controller/UserController.java]: Add

**Review by IllyaZuew — APPROVED** [review]

**Review by Sershok — APPROVED** [review]

**Review by rpkyrych — APPROVED** [review]
Looks good!

**Review by demus-n — APPROVED** [review]

**Review by ennnbaev — APPROVED** [review]

**Pull request merged**